src/mame/sony/news_68k_iop.cpp

```diff
@@ -0,0 +1,1150 @@
+ * There are a couple of mentions of the Hyperbus in NEWS-OS 4 related to even the NWS-800, but the NWS-1xxx/3xxx series were already released by the time NEWS-OS 4 was.
+ * So, for clarity due to the number of busses on this system, the Hyperbus term is used throughout this code, but it may or may not be a 100% accurate description.
+ * The Hyperbus is directly connected to main memory, the system ROM, and the VME bus interface. Everything else is connected to the I/O
+ * bus. The IOP, CPU/MMU, and VME bus can all access the Hyperbus through a buffer. As far as I can tell, nothing on the Hyperbus
+ * can talk to the I/O bus, only the IOP is able to do that.
+ *
+ *  Supported:
+ * 	 - NWS-831
+ *
+ *  Not supported yet:
+ *   - All other NWS-8xx workstations
+ *   - NWS-9xx servers (2x '020)
+ *
+ * Known NWS-800 Series Base Configurations
+ * - NWS-811: Jun 1987, 4MB RAM, no cache, diskless
+ * - NWS-820: Jan 1987, 4MB RAM, ?? cache, 86MB HDD
+ * - NWS-821: Aug 1987, 4MB RAM, no cache, 156MB HDD
+ * - NWS-830: Jan 1987, 8MB RAM (expandable?), ?? cache, 156MB HDD
+ * - NWS-831: Sep 1987, 8MB RAM (up to 16MB), 8KB cache, 156MB HDD
+ * - NWS-841: Sep 1987, 8MB RAM (up to 16MB), 8KB cache, 256MB HDD
+ * - NWS-891: Apr 1988, 4MB RAM, no cache, 86MB HDD, with CD-ROM drive
+ *
+ * Known NWS-900 Series Base Configurations:
+ * - NWS-911: Feb 1988, 2x 286MB HDD
+ * - NWS-921: Feb 1988, 4x 286MB HDD
+ *
+ * References:
+ *  - NWS-1960 Service Guide
+ *  - http://bitsavers.org/pdf/sony/news/Sony_NEWS_Technical_Manual_3ed_199103.pdf
+ *  - https://katsu.watanabe.name/doc/sonynews/model.html
+ *
+ * TODO:
+ *   - MMU emulation improvements (are all the right status bits set? any missing features? etc)
+ *   - Debug general OS issues - random segfaults when running `ps`, system sometimes fails to shutdown when running `shutdown -x now` after using the networking stack, etc.
+ *   - AST (Asynchronous System Trap) emulation
+ *   - System cache emulation
+ *   - Expansion slots (I/O Bus and VMEBus)
+ *   - Networking is very flaky, especially on NEWS-OS 4.
+ *   - Graphics, kbms, and parallel port emulation
+ *   - Hyperbus handshake for IOP and CPU accesses. The bus has arbitration circuitry to prevent bus contention when both the CPU and IOP are trying to access the hyperbus (RAM and VME)
+ */
+
+#include "emu.h"
+
+#include "news_hid.h"
+#include "news_020_mmu.h"
+#include "news_iop_scsi.h"
+
+#include "bus/nscsi/cd.h"
+#include "bus/nscsi/hd.h"
+#include "bus/nscsi/tape.h"
+#include "bus/rs232/rs232.h"
+
+#include "cpu/m68000/m68020.h"
+
+#include "formats/pc_dsk.h"
+
+#include "imagedev/floppy.h"
+
+#include "machine/am79c90.h"
+#include "machine/msm58321.h"
+#include "machine/ncr5380.h"
+#include "machine/nscsi_bus.h"
+#include "machine/pit8253.h"
+#include "machine/ram.h"
+#include "machine/timekpr.h"
+#include "machine/upd765.h"
+#include "machine/z80scc.h"
+#include "machine/bankdev.h"
```

Comment:
The stuff from `formats` is supposed to come after the stuff from `devices`.  Also, please sort the headers within each block.